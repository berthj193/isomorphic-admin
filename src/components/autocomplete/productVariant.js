import React from 'react';
import PropTypes from 'prop-types';
import withStoresManager from '../apolloProviders/withStoresManager';
import productVariantQueries from '../../graphql/productVariants';
import Select, { SelectOption } from '../uielements/select';

export class ProductVariantAutocomplete extends React.Component {
  static propTypes = {
    product_variants: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string,
      default_sku: PropTypes.string,
    })),
  };

  static defaultProps = {
    product_variants: [],
  };

  dataSource = (this.props.product_variants || []).map(({ id, default_sku }) =>
    <SelectOption value={id} key={id}>{`${default_sku}`}</SelectOption>);

  render() {
    return (
      <Select
        showSearch
        {...this.props}
      >
        {this.dataSource}
      </Select>
    );
  }
}

export default withStoresManager(ProductVariantAutocomplete, [
  {
    query: productVariantQueries.productVariantAutocomplete,
  },
]);
