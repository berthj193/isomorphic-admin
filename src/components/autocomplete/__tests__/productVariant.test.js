import React from 'react';
import { ProductVariantAutocomplete } from '../productVariant';
import { shallow } from 'enzyme';

describe('product variants autocomplete component', () => {
  const getComponent = (props = {}) => shallow(<ProductVariantAutocomplete {...props} />);
  const dataObj = {
    id: '1',
    name: 'foo',
  };

  it('datasource should return empty array when there are no elements', () => {
    const instance = getComponent().instance();
    expect(instance.dataSource.length).toBe(0);
  });

  it('datasource should return one element', () => {
    const instance = getComponent({
      product_variants: [dataObj],
    }).instance();

    expect(instance.dataSource).toMatchSnapshot();
  });
});
