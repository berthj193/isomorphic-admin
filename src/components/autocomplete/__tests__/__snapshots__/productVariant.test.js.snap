// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product variants autocomplete component datasource should return one element 1`] = `
Array [
  <Option
    value="1"
>
    undefined
</Option>,
]
`;
